### PR TITLE
gcp - pub sub resources get iam policy

### PR DESCRIPTION
**Pull request by alexkarpitski**
Use the same approach as there:
https://github.com/mediapills/cloud-custodian/pull/19

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/pubsub.py [hunk 1]
```diff
@@ -36,6 +36,24 @@ def get(client, resource_info):
                 'get', {'topic': resource_info['topic_id']})
 
 
+@resources.register('pubsub-topic-iam-policy')
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/pubsub.py]: this is not a new resource. it is an action on an existing resource and needs to be registered as such. the class structure is also different. see the `c7n_gcp/resources/sql.py` for example of this.

you'll need to see if there's any methods you need to overwrite in the `c7n_gcp.actions.core.MethodAction` class.

Comment (jondavidi) [reply tools/c7n_gcp/c7n_gcp/resources/pubsub.py]: I can see there's at least 2 methods that will need overwriting if you look at the MethodAction class it's indicated there.

File: tools/c7n_gcp/c7n_gcp/resources/pubsub.py [hunk 2]
```diff
@@ -53,12 +71,49 @@ def get(client, resource_info):
                 'get', {'subscription': resource_info['name']})
 
 
+@resources.register('pubsub-subscription-iam-policy')
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/pubsub.py]: same thing here

File: tools/c7n_gcp/c7n_gcp/resources/pubsub.py [hunk 3]
```diff
@@ -53,12 +71,49 @@ def get(client, resource_info):
...
     class resource_type(TypeInfo):
         service = 'pubsub'
         version = 'v1'
         component = 'projects.snapshots'
         enum_spec = ('list', 'snapshots[]', None)
         scope_template = 'projects/{}'
         id = 'name'
+
+
+@resources.register('pubsub-snapshot-iam-policy')
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/pubsub.py]: same thing here

**Comment by alexkarpitski** [comment]
Ok, I see, thank you very much for the comments, it will be reworked once we get to implementing actions for the resources.

**Pull request closed without merging**